water_mark/etl_water_mark.py

```diff
@@ -0,0 +1,142 @@
+####################                            ###############################
+################### George Meier 16 July 2019  ################################
+##################  mater_mark ETL routine    #################################
+########################################################################80chars
+#
+# This program sets up the current directory for water_mark to be run by
+# creating an input directory, and filling it with VV and  VH sar tifs
+# downloaded from asf HyP3.
+#
+###############################################################################
+# Python3
+# Windows admin powershell or Linux
+# '-vrt True' to make vrt
+# - water_mark:
+#   - inputs: (input directory made automatically)
+#   - water_mark.py
+#   - etl_water_mark.py
+#   - download-all-<nums>.py (python script from asf HyP3)
+#   - gdal_reclassify.py   (needed for vrt)
+#   - downloadWaterData.py (needed for vrt)
+################################################################################
+
+import argparse
+import platform
+import os
+import shutil
+import subprocess
+import zipfile
+from typing import Tuple
+
+
+def determine_OS() -> Tuple[bool, bool]:
+    if platform.system() == 'Windows':
+        return (True, False)
+    else:
+        return (False, True)
+
+
+def get_SAR_from_HyP3() -> None:
+    linuxMode, windowsMode = determine_OS()
+    python = 'python'
+    if linuxMode:
+        python = 'python3'
+    scriptToRun = 'temp'
+    for fileName in os.listdir():
+        if 'download' in fileName:
+            scriptToRun = fileName
+    subprocess.call(f"{python} {scriptToRun}",
+                    shell=True)
+
+
+def make_input_dir() -> None:
+    inputsPath = os.path.join(os.getcwd(), 'inputs')
+    if not os.path.exists(inputsPath):
+        os.mkdir(inputsPath)
+        if os.path.exists(os.path.join(os.getcwd(), 'gdal_reclassify.py')):
+            shutil.move(os.path.join(os.getcwd(), 'gdal_reclassify.py'),
+                        inputsPath)
+        if os.path.exists(os.path.join(os.getcwd(), 'downloadWaterData.py')):
+            shutil.move(os.path.join(os.getcwd(), 'downloadWaterData.py'),
+                        inputsPath)
+
+
+def make_vrt() -> None:
+    inputsPath = os.path.join(os.getcwd(), 'inputs')
+    worldMaskPath = os.path.join(inputsPath, 'worldMask')
+    if os.path.exists(worldMaskPath):
+        shutil.rmtree(worldMaskPath)
+    os.mkdir(worldMaskPath)
+    windowsMode, linuxMode = determine_OS()
+    python = 'python'
+    if linuxMode:
+        python = 'python3'
+    scriptPath = os.path.join(inputsPath, 'downloadWaterData.py')
+    subprocess.call(f"{python} {scriptPath} -d '{worldMaskPath}' occurrence",
+                    shell=True)
+    with open(os.path.join(worldMaskPath, 'worldMask.txt'), 'w') as f:
+        for ff in os.listdir(worldMaskPath):
+            f.write(ff+'\n')
+    subprocess.call(f"gdalbuildvrt -input_file_list \
```

Comment:
use python bindings: 
https://gdal.org/python/osgeo.gdal-module.html#BuildVRT

Reply:
This one I'm not going to change because the link you provided dosent have the documentation needed 
also
https://gis.stackexchange.com/questions/81788/gdalbuildvrt-error-when-using-in-python 
it seems my way is not unheard of